kwant_to_openfermion.py

```diff
@@ -17,25 +67,37 @@ def system_to_FermionOperator(sys):
         The hamiltonian of sys as an openfermion object.
     '''
     
-    if not isinstance(sys, kwant.system.System):
-        raise TypeError(f'Expecting an instance of System, got {type(sys)}')
+    
+    #Get the number of spin states out of the first on-site value
+    sample_val = sys.onsites[0][0]
+    if isinstance(sample_val, (int, float, complex)):
+        n_spin = 1
+    elif isinstance(sample_val, (numpy.ndarray, tinyarray.ndarray_complex, tinyarray.ndarray_float, tinyarray.ndarray_int)):
```

Comment:
It might be more pythonic to do this check with a 'try, except' statement, instead of explicitly type-checking. Also, check the length of your lines here.